react-events-and-state/src/index.jsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class CustomButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
+    this.state = { isClicked: false };
+  }
+
+  handleClick() {
+    this.setState({ isClicked: true });
+  }
+
+  render() {
+    if (this.state.isClicked) {
+      this.props.text = 'Thanks Buddy';
+    } else {
+      this.props.text = 'Click Me!';
+    }
```

Comment:
We never assign to props in React. Instead declare a `let` variable and assign it a value in your conditional logic.

```js
let text;
if (something) {
  text = '??';
} else {
  text = '??';
}
```